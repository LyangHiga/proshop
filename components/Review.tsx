import StarIcon from "@material-ui/icons/Star";
import StarHalfIcon from "@material-ui/icons/StarHalf";
import StarOutlineIcon from "@material-ui/icons/StarOutline";

interface ReviewProps {
  stars: number;
}

const Review = (props: ReviewProps) => {
  const numberOfStars = Number.isInteger(props.stars)
    ? props.stars
    : props.stars - 0.5;
  const halfStar = Number.isInteger(props.stars) ? null : (
    <StarHalfIcon style={{ color: "#f6e336" }} />
  );
  const emptyStars = halfStar ? 4 - numberOfStars : 5 - numberOfStars;

  return (
    <div>
      {[...Array(numberOfStars)].map((e, i) => (
        <StarIcon key={i} style={{ color: "#f6e336" }} />
      ))}
      {halfStar}
      {props.stars > 4
        ? null
        : [...Array(emptyStars)].map((e, i) => (
            <StarOutlineIcon key={i} style={{ color: "#f6e336" }} />
          ))}
    </div>
  );
};

export default Review;
