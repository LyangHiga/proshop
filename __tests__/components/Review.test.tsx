import React from "react";
import { shallow } from "enzyme";
import Review from "../../components/Review";
describe("Review", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Review stars={4} numberReviews={10} />);
    expect(wrapper).toMatchSnapshot();
  });
});
