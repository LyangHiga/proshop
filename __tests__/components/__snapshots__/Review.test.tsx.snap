// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review renders correctly 1`] = `ShallowWrapper {}`;
